### ISSUE TITLE
Background maintenance taking too long and creating read spikes on DynamoDB

### ISSUE THREAD

**Issue by raunyoliveira-hotmart** [silinternational/certmagic-storage-dynamodb#20]
Hello there,

First, thank you for creating and maintaining this plugin!

I'm using this module with Caddy version 2.4.6 running inside Kubernetes with 3 replicas. 
It's working great, serving 25k+ certificates, with low resources usage for cpu and memory. 😄 

The issue I'm having is when the `background maintenance task` starts. 
It's taking to long to finish and it creates spikes of read on DynamoDB for the duration of the background maintenance task.

On normal usage reads on DynamoDB are around 700 per minute, but we are seeing spikes going up to 40k.
The last maintenance task took 14hs to finish, and the read was in 40k for the whole time.

Are there any suggestions or workarounds for this issue?
1. I've posted this problem on [Caddy forum](https://caddy.community/t/caddy-dynamodb-background-maintenance-taking-too-long-and-creating-read-spikes-on-dynamodb/14153/2?u=rauny), and they recommended switching to Redis.
2. I'm thinking on running fewer replicas, because based on the logs, each replica is scanning the DynamoDB table. Is this correct, I'm not sure about this?
3. Any other ideas or suggestions?

Thank you!

**Dockerfile**

```
FROM caddy:2.4.6-builder AS builder

RUN xcaddy build \
    --with github.com/silinternational/certmagic-storage-dynamodb

FROM caddy:2.4.6-alpine

COPY --from=builder /usr/bin/caddy /usr/bin/caddy
```

**Caddyfile**

```
{
	on_demand_tls {
		ask https://check-domain.internal.endpoint/cname
		interval 1m
		burst 200
	}
	storage_clean_interval 90d
	storage dynamodb caddy-certificates {
		aws_region us-east-1
	}
}

https://
tls {
	on_demand
	issuer zerossl <key> {
		email <email>
		timeout 3m
	}
	issuer acme {
		email <email>
		timeout 3m
	}
}
reverse_proxy hostingapp
```

**Comment by raunyoliveira-hotmart** [comment]
[https://caddy.community/t/caddy-dynamodb-background-maintenance-taking-too-long-and-creating-read-spikes-on-dynamodb/14153/8?u=rauny](https://caddy.community/t/caddy-dynamodb-background-maintenance-taking-too-long-and-creating-read-spikes-on-dynamodb/14153/8?u=rauny)

**Comment by raunyoliveira-hotmart** [comment]
fix for this problem:

https://caddy.community/t/caddy-dynamodb-background-maintenance-taking-too-long-and-creating-read-spikes-on-dynamodb/14153/9

TL;DR:
switch storage to file system with JSON config and enable `renew_interval` (at this time this config is only available in JSON)
